<launch>
    <!-- <rosparam param="sim">false</rosparam>
    <rosparam param="map">true</rosparam>
    <rosparam param="rviz">true</rosparam>
    <rosparam param="use_tf">true</rosparam> -->

  <!-- <node pkg="gmapping" type="slam_gmapping" name="turtlebot3_slam_gmapping" output="log">
    <param name="base_frame" value="base_footprint"/>
    <param name="odom_frame" value="odom"/>
    <param name="map_update_interval" value="1.0"/>
    <param name="maxUrange" value="4.0"/>
    <param name="minimumScore" value="100"/>
    <param name="linearUpdate" value="0.2"/>
    <param name="angularUpdate" value="0.2"/>
    <param name="temporalUpdate" value="0.5"/>
    <param name="delta" value="0.05"/>
    <param name="lskip" value="0"/>
    <param name="particles" value="120"/>
    <param name="sigma" value="0.05"/>
    <param name="kernelSize" value="1"/>
    <param name="lstep" value="0.05"/>
    <param name="astep" value="0.05"/>
    <param name="iterations" value="5"/>
    <param name="lsigma" value="0.075"/>
    <param name="ogain" value="3.0"/>
    <param name="srr" value="0.01"/>
    <param name="srt" value="0.02"/>
    <param name="str" value="0.01"/>
    <param name="stt" value="0.02"/>
    <param name="resampleThreshold" value="0.5"/>
    <param name="xmin" value="-10.0"/>
    <param name="ymin" value="-10.0"/>
    <param name="xmax" value="10.0"/>
    <param name="ymax" value="10.0"/>
    <param name="llsamplerange" value="0.01"/>
    <param name="llsamplestep" value="0.01"/>
    <param name="lasamplerange" value="0.005"/>
    <param name="lasamplestep" value="0.005"/>
  </node> -->


  <!-- TO DO: FIX CONTROLLER/PARAMS -->
  <rosparam param="sim">false</rosparam>
  <rosparam param="map">true</rosparam>
  <rosparam param="rviz">true</rosparam>
  <rosparam param="use_tf">true</rosparam>
  <node pkg="asl_turtlebot" type="supervisor_nav.py" name="turtlebot_supervisor_nav" output="screen"/>
  <node pkg="asl_turtlebot" type="detector.py" name="turtlebot_detector" output="screen"/>
  <node pkg="asl_turtlebot" type="pose_controller_nav.py" name="turtlebot_pose_controller_nav" />
  <node pkg="asl_turtlebot" type="navigator.py" name="turtlebot_navigator" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find asl_turtlebot)/rviz/nav.rviz"/>

  <!-- <node pkg="tf" type="static_transform_publisher" name="camera_true_broadcaster" args="0.03 0 0.067 0.707107 0 0.707107 0 base_footprint raspicam 100" />

  <node pkg="tf" type="static_transform_publisher" name="camera_broadcaster" args="0 0 0 0 0 0.707107 0.707107 raspicam camera 100" />

  <node pkg="tf" type="static_transform_publisher" name="velodyne_broadcaster" args="-0.04 0 0.215 0 0 0 1 base_footprint velodyne 100" /> -->
</launch>
